src/planscape/forsys/forsys_request_params.py

```diff
@@ -0,0 +1,294 @@
+                     (-120.18409937110482, 38.48622140686506),
+                     (-119.93422142411087, 38.48622140686506),
+                     (-119.93422142411087, 38.05413814388948),
+                     (-120.14015536869722, 38.05413814388948)))
+        p2.srid = srid
+        p3 = Polygon(((-121.14015536869722, 39.05413814388948),
+                     (-121.18409937110482, 39.48622140686506),
+                     (-120.53422142411087, 39.48622140686506),
+                     (-120.53422142411087, 39.05413814388948),
+                     (-121.14015536869722, 39.05413814388948)))
+        p3.srid = srid
+        m1 = MultiPolygon(p1, p2)
+        m1.srid = srid
+        m2 = MultiPolygon(p3)
+        m2.srid = srid
+        return {1: m1,
+                2: m2}
+
+    def _read_project_areas_from_url_params(
+            self, params: QueryDict) -> dict[int, MultiPolygon]:
+        project_areas = {}
+        for project_area_str in params.getlist(self._URL_PROJECT_AREAS):
+            geo = GeoFromUrlParams(
+                json.loads(project_area_str))
+            multipolygon = _transform_geo_from_url_params_into_multipolygon(
+                geo, self._URL_PROJECT_AREAS)
+            project_areas[geo['id']] = multipolygon
+        return project_areas
+
+
+# Gathers forsys input parameters from url params, database lookups, or a
+# combination of the two.
+# Of note, the option to set all forsys input paramters via url parameters is
+# intended for backend debugging purposes while the option to set most forsys
+# input parameters via database lookups is intended for production.
+class ForsysProjectAreaGenerationRequestParams():
+    # Constants for parsing url parameters.
+    _URL_USE_ONLY_URL_PARAMS = 'set_all_params_via_url_with_default_values'
+    _URL_REGION = 'region'
+    _URL_PRIORITIES = 'priorities'
+    _URL_PRIORITY_WEIGHTS = 'priority_weights'
+    _URL_PLANNING_AREA = 'planning_area'
+
+    # Constants that act as default values when parsing url parameters.
+    _DEFAULT_REGION = 'sierra_cascade_inyo'
+    _DEFAULT_PRIORITIES = ['fire_dynamics',
+                           'forest_resilience', 'species_diversity']
+
+    # TODO: make regions and priorities enums to make error checking easier.
+    # TODO: add fields for costs, treatments, and global, project-level, and
+    # stand-level constraints.
+    # The planning region.
+    region: str
+    # Conditions whose AP scores will be considered when ranking projects.
+    priorities: list[str]
+    # Weights associated with the priorities.
+    # Must be same length as priorities.
+    # If not set explicitly, this is a vector of 1's by default.
+    priority_weights: list[float]
+    # Project areas to be ranked. A project area may consist of multiple
+    # disjoint polygons. The dict is keyed by project ID.
+    planning_area: MultiPolygon
+
+    def __init__(self, request: HttpRequest) -> None:
+        params = request.POST
```

Comment:
this line doesn't exist in the function above. they should probably use the same HTTP method, so let's keep them consistent

Reply:
They should use the same method, but that's a TODO for later.

I set HttpRequest as input to the constructor because I need request.user to retrieve a Plan. I haven't made the same change to the RankingRequest because 1) it's not in the scope of this pr and 2) we haven't settled on db models so I'm unsure what permissions for retrieving projects is like and I anticipate changing it again (and settling on that is out of scope of this pr).